client/packages/system/src/Patient/Encounter/ListView.tsx

```diff
@@ -70,15 +63,7 @@ const EncounterListComponent: FC = () => {
 };
 
 export const EncounterListView: FC = () => (
-  <TableProvider
-    createStore={createTableStore}
-    queryParamsStore={createQueryParamsStore<EncounterFragmentWithStatus>({
-      initialSortBy: {
-        key: EncounterSortFieldInput.StartDatetime,
-        isDesc: true,
-      },
-    })}
-  >
```

Comment:
Same as above.

Reply:
The problem here is that if you select a sort option and change tabs back end forth to the encounter list you would expect that the sort/filter options are remembered. With these changes they are reverted to the default though. This was the reason why there were store and the url query params...

I think what we want is a way to store the filter/sort params in the url AND in the store and when you revisit a tab the url params are initialized from the store params. I would suggest creating a new hook for this...
@mark-prins what do you think?

Reply:
Yeah, that could work. I'd be a little concerned about how difficult that could be, and we'd need to write out requirements and test cases; we've spent so much time fixing the URL query params already.. it's a complex area.

Reply:
Yes, happy to move it to another issue.

@richardmoizeau how much do we want to remember encounter filter/sort options when switching between tabs in the patient view?

Reply:
I don't there is any filter for the patient view right ? I think that's OK not to save the sort order (would definitely be nice though).

Reply:
I tried to look at the `useQueryParamsStore` hook thinking that it supposed to work or persist the last sorted key and its direction (of which it is being made for) at first place. But the problem I can see is the `useQueryParamsStore` is being used with state initialized as a props from the `TableProvider`. The patient encounter `EncounterListView` component gets remounted every time the tab gets changed resulting the reset of all states of `useQueryParamsStore` to its initialized props value. If we separate the `useQueryParamsStore` usage from `TableProvider` to external store then no sorting would work at all cause the `useTableStore` hook is being used to handle all datatable action and is the consumer of `TableProvider` so need to be subscribed under `TableProvider` component to function all table actions. The only option I can see working is to make `useQueryParamsStore` persistence using `localStorage`. But this would give another complication to handle mount/unmount of `EncounterListView` component to reset sorting to its default state.

So I'm also inclining towards not to implement this sort persistence in the datatable (unless user specifically asks for).

Reply:
agree - that's getting too complex for this issue. Something to consider another time perhaps.